Autopilot/PathManager/Inc/cruisingState.hpp

```diff
@@ -0,0 +1,37 @@
+/*
+* Declares the functions that interact with the Waypoint Manager (on behalf of the cruisingState state)
+* Author: Dhruv Rawat
+*/
+
+#include "pathDatatypes.hpp"
+#include "waypointManager.hpp"
+
+#ifndef CRUISING_STATE
+#define CRUISING_STATE
+
+struct _CruisingState_Telemetry_Return {
+    char editingFlightPathErrorCode; // 0 = success, 1 = error, 2 = incorrect telemetry command
+    char pathFollowingErrorCode; // 0 = success, 1 = error, 2 = home base not initialized, 3 = incorrect telemetry command
+    int currentWaypointId; 
+    int currentWaypointIndex;
+    bool homeBaseInitialized;
+};
+
+// The following functions are used to interact with the Waypoint Manager module
+int editFlightPath(Telemetry_PIGO_t * telemetryData, WaypointManager& cruisingStateManager, int * idArray);
+
+int pathFollow(Telemetry_PIGO_t * telemetryData, WaypointManager& cruisingStateManager, _WaypointManager_Data_In input, _WaypointManager_Data_Out * output, bool& goingHome, bool& inHold);
```

Comment:
Take a second to give each of these functions a description and to explain what the arguments and return values are